AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_DeferredFogAdded.py

```diff
@@ -111,78 +185,236 @@ def AtomEditorComponents_DeferredFog_AddedToEntity():
+                      deferred_fog_component.get_component_property_value(
+                          AtomComponentProperties.deferred_fog('Enable Fog Layer')) is True)
+
+        # Disable the Enable Fog Layer parameter.
+        deferred_fog_component.set_component_property_value(
+            AtomComponentProperties.deferred_fog('Enable Fog Layer'), False)
+        Report.result(Tests.enable_fog_layer_parameter_disabled,
+                      deferred_fog_component.get_component_property_value(
+                          AtomComponentProperties.deferred_fog('Enable Fog Layer')) is False)
+
+        # Re-enable the Enable Fog Layer parameter for game mode verification.
+        deferred_fog_component.set_component_property_value(
+            AtomComponentProperties.deferred_fog('Enable Fog Layer'), True)
+        general.idle_wait_frames(1)
+
+        # 11. Edit the Fog Color parameter.
+        violet_color_value = math.Vector3(0.498, 0.0, 1.0)
+        deferred_fog_component.set_component_property_value(
+            AtomComponentProperties.deferred_fog('Fog Color'), violet_color_value)
+        fog_color_value = deferred_fog_component.get_component_property_value(
+            AtomComponentProperties.deferred_fog('Fog Color'))
+        Report.result(Tests.edit_fog_color, fog_color_value.IsClose(violet_color_value))
+
+        # 12. Update the Fog Start Distance parameter to min/max values.
+        # Update the Fog Start Distance parameter to its minimum value.
+        deferred_fog_component.set_component_property_value(
+            AtomComponentProperties.deferred_fog('Fog Start Distance'), 0.0)
+        Report.result(Tests.fog_start_min,
+                      deferred_fog_component.get_component_property_value(
+                          AtomComponentProperties.deferred_fog('Fog Start Distance')) == 0.0)
+
+        # Update the Fog Start Distance parameter to its maximum value.
+        deferred_fog_component.set_component_property_value(
+            AtomComponentProperties.deferred_fog('Fog Start Distance'), 5000.0)
+        Report.result(Tests.fog_start_max,
+                      deferred_fog_component.get_component_property_value(
+                          AtomComponentProperties.deferred_fog('Fog Start Distance')) == 5000.0)
+
+        # 13. Update the Fog End Distance parameter to min/max values.
+        # Update the Fog End Distance parameter to its maximum value.
+        deferred_fog_component.set_component_property_value(
+            AtomComponentProperties.deferred_fog('Fog End Distance'), 5000.0)
+        Report.result(Tests.fog_end_max,
+                      deferred_fog_component.get_component_property_value(
+                          AtomComponentProperties.deferred_fog('Fog End Distance')) == 5000.0)
+
+        # Update the Fog End Distance parameter to its minimum value.
+        deferred_fog_component.set_component_property_value(
+            AtomComponentProperties.deferred_fog('Fog End Distance'), 0.0)
```

Comment:
is there anything that prevents you from setting the end distance to zero while the start distance is 5000? it seems like it should but these getter/setters seem to be fine with taking anything in individual value range.

Reply:
Nope, you're free to set them to whatever you want. There's nothing that stops you from setting start distance higher than end distance. Currently its up to the user to know what to do.